docs: add `ModifyColumn` on nightly (#935)
GreptimeTeam/docs: docs/nightly/en/reference/sql/alter.md

The removed column can't be retrieved immediately by all subsequent queries.

### Modify column type

> Modify the date type of column

Reviewer: ```suggestion
Modify the date type of a column
```

```sql
ALTER TABLE monitor MODIFY COLUMN load_15 STRING;
```

> The modified column cannot be a tag (primary key) or time index, and it must be nullable to ensure that the data can be safely converted (become null).

Reviewer: ```suggestion
The modified column cannot be a tag (primary key) or time index, and it must be nullable to ensure that the data can be safely converted (returns null on cast failures).
```

> The modified column cannot be a tag (primary key) or timestamp (time index), and it must be Nullable to ensure that the data can be safely converted (become null).

Reviewer: ```suggestion
The modified column cannot be a tag (primary key) or time index, and it must be nullable to ensure that the data can be safely converted (become null).
```

> The modified column cannot be a tag (primary key) or time index, and it must be nullable to ensure that the data can be safely converted (returns null on cast failures).

Reviewer 2: ```suggestion
The modified column cannot be a tag (primary key) or time index, and it must be nullable to ensure that the data can be safely converted (returns `NULL` on cast failures).
```
